### PR TITLE
feature: Add new tracker get endpoint test

### PR DESCRIPTION
**Pull request by muilpp**
This PR aims to create a new test to validate the new Tracker API get endpoint and do a bit of refactor with the old API tests.

### CODE REVIEW

File: src/main/java/org/hisp/dhis/tasks/tracker/tei/GetEntitiesTask.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.hisp.dhis.actions.AuthenticatedApiActions;
+import org.hisp.dhis.cache.UserCredentials;
+import org.hisp.dhis.response.dto.ApiResponse;
+import org.hisp.dhis.tasks.DhisAbstractTask;
+import org.hisp.dhis.utils.Randomizer;
+
+/**
+ * @author Marc Pratllusa <marc@dhis2.org>
+ */
+public class GetEntitiesTask
```
Comment (enricocolasante) [inline src/main/java/org/hisp/dhis/tasks/tracker/tei/GetEntitiesTask.java:12]: Following the pattern in the package I would name it `GetTeisTask`

File: src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            String teiId = row.get( 0 ).toString();
+
+            new GetTeiTask( teiId, user.getUserCredentials(), rnd ).execute();
+        }
+
+        waitBetweenTasks(rnd);
+
+    }
+
+    private String getAttributesQuery(Program program, Randomizer rnd)
```
Comment (enricocolasante) [inline src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java:75]: Is this and the following private methods ever used?

File: src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        if ( rows != null && !rows.isEmpty() )
+        {
+            ArrayList row =rnd.randomElementFromList( rows );
+
+            String teiId = row.get( 0 ).toString();
+
+            new GetTeiTask( teiId, user.getUserCredentials(), rnd ).execute();
+        }
+
```
Comment (enricocolasante) [inline src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java:70]: I am not sure I understand why this endpoint call is here.
I know we are doing something similar in other taskset, but if this is not something that we want to be there for a purpose, I would suggest to remove it in order to have the simplest and clearest test as possible

**Review by enricocolasante — COMMENTED** [review]
Nice job!
It is good to see that we can add new performance tests to the suite in an (almost) easy way.
I added just a few suggestions

File: src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java (hunk 3 above)
Comment (muilpp) [reply src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java]: Now I tried to simplified as much as possible ;)

**Review by enricocolasante — CHANGES_REQUESTED** [review]
Sorry to come back with more comments but I opened the PR in the IDE and the package organisation is quite messy and moving stuff to oldapi folder can confuse things even more.
Not sure why we ended up with this convention but `tracker/importer` is the "new" tracker.
Everything in tei, enrollmentss and event folder under task folder is old api.
My suggestion for this PR would be to not move things around, put GetTeiTask.java (it actually should be GetTeisTask.java as it can return multiple teis) under task/tracker/importer and TrackerCapture_getEntitiesOrderedByEnrolledAt.java under trackerset/tracker/importer.

Then maybe we can have a look at it together and find a good package naming for old and new api

**Review by enricocolasante — APPROVED** [review]
I think it makes much more sense now :100:

**Review by teleivo — APPROVED** [review]

File: src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * @author Marc Pratllusa <marc@dhis2.org>
+ */
+public class TrackerCapture_getEntitiesOrderedByEnrolledAt
+    extends DhisAbstractTaskSet
+{
+    private static final String NAME = "TrackerCapture: get entities ordered by enrolment date";
```
[the remark this answers is not in the document]
Comment (muilpp) [reply src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java:15]: that's something that came up already, enrolment is British and enrollment is American, same as organization and organisation (which I had to change in another PR), do we use American English only?

[the remark this answers is not in the document]
Comment (teleivo) [reply src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java:15]: I think you would need to ask on Slack maybe the backend channel or even the core-dev one about that.

What I can say is that I would stick to what we are usually doing. So I would search the code and then look at the results
https://github.com/dhis2/dhis2-core/search?q=enrolment
https://github.com/dhis2/dhis2-core/search?q=Enrolment
https://github.com/dhis2/dhis2-core/search?q=enrollment
https://github.com/dhis2/dhis2-core/search?q=Enrollment

at least for enrollment its pretty clear that we use `enrollment` almost exclusively.

**Review by teleivo — APPROVED** [review]

**Pull request merged**